packages/compiler/lib/decorators.ts

```diff
@@ -40,10 +40,34 @@ export function $doc(
     text = replaceTemplatedStringFromProperties(text, sourceObject);
   }
 
-  program.stateMap(docsKey).set(target, text);
+  program.stateMap(key).set(target, text);
+}
+
+const summaryKey = Symbol();
+export function $summary(
```

Comment:
Just a note that if this get merged after the [validation and tests pr](https://github.com/microsoft/cadl/pull/236) to just add the same validation and tests as `@doc`

Reply:
That's not merged yet, but I can certainly follow up after it to add it.